$NetBSD: patch-ab,v 1.2 2005/12/19 00:06:47 joerg Exp $

Use <strings.h>'s index() instead of a K&R declaration with another return
type, which conflicts wherever <string.h> brings <strings.h> in (glibc, and
clang's IRIX wrapper headers).

--- decode.c.orig
+++ decode.c
@@ -16,6 +16,9 @@
 #include <time.h>
 #define index   strchr
 #else
+#ifdef __linux__
+#include <time.h>
+#endif
 #ifdef __TURBOC__
 #include <time.h>
 #include <io.h>
@@ -37,8 +40,7 @@
 int  type,dline;
 unsigned char decbuf[LBUFLEN][LBUFLEN];
 
-extern unsigned char *index();
-extern int errno;
+#include <strings.h>	/* index(); was a K&R declaration */
 
 decode()
 {
